Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -1261,36 +1262,32 @@ static inline bool shouldAddContentKeyRecipients()
 void SourceBufferPrivateAVFObjC::configureVideoRenderer(VideoMediaSampleRenderer& videoRenderer)
 {
     videoRenderer.setResourceOwner(m_resourceOwner);
-    RetainPtr displayLayer = dynamic_objc_cast<AVSampleBufferDisplayLayer>(videoRenderer.renderer());
 #if ENABLE(ENCRYPTED_MEDIA) && HAVE(AVCONTENTKEYSESSION)
-    if (m_cdmInstance && shouldAddContentKeyRecipients() && displayLayer)
-        [m_cdmInstance->contentKeySession() addContentKeyRecipient:displayLayer.get()];
+    if (m_cdmInstance && shouldAddContentKeyRecipients() && videoRenderer.as<AVSampleBufferDisplayLayer>())
+        [m_cdmInstance->contentKeySession() addContentKeyRecipient:videoRenderer.as<AVSampleBufferDisplayLayer>()];
 #endif
     videoRenderer.requestMediaDataWhenReady([weakThis = ThreadSafeWeakPtr { *this }] {
         if (RefPtr protectedThis = weakThis.get(); protectedThis && protectedThis->m_enabledVideoTrackID)
             protectedThis->didBecomeReadyForMoreSamples(*protectedThis->m_enabledVideoTrackID);
     });
-    if (displayLayer)
-        m_listener->beginObservingVideoRenderer(displayLayer.get());
+    m_listener->beginObservingVideoRenderer(videoRenderer.renderer());
 }
 
 void SourceBufferPrivateAVFObjC::invalidateVideoRenderer(VideoMediaSampleRenderer& videoRenderer)
 {
     videoRenderer.flush();
     videoRenderer.stopRequestingMediaData();
-    if (RetainPtr renderer = videoRenderer.renderer())
-        m_listener->stopObservingVideoRenderer(renderer.get());
+    m_listener->stopObservingVideoRenderer(videoRenderer.renderer());
 
 #if ENABLE(ENCRYPTED_MEDIA) && HAVE(AVCONTENTKEYSESSION)
-    RetainPtr displayLayer = dynamic_objc_cast<AVSampleBufferDisplayLayer>(videoRenderer.renderer());
-    if (displayLayer && m_cdmInstance && shouldAddContentKeyRecipients())
-        [m_cdmInstance->contentKeySession() removeContentKeyRecipient:displayLayer.get()];
+    if (m_cdmInstance && shouldAddContentKeyRecipients() && videoRenderer.as<AVSampleBufferDisplayLayer>())
+        [m_cdmInstance->contentKeySession() removeContentKeyRecipient:videoRenderer.as<AVSampleBufferDisplayLayer>()];
 #endif
 }
 
 void SourceBufferPrivateAVFObjC::setVideoRenderer(VideoMediaSampleRenderer* renderer)
 {
-    if (renderer == m_videoRenderer) {
+    if (m_videoRenderer && m_videoRenderer == renderer) {
```

Comment:
What does this change achieve?

Reply:
This is the original code once you include the revert.

https://searchfox.org/wubkat/rev/c3d6184dbe8d3daa73c7c9201ce556cf81e35d3e/Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm#1324-1327 

```
    if (m_videoRenderer && renderer == m_videoRenderer->renderer()) {
        if (RefPtr expiringVideoRenderer = std::exchange(m_expiringVideoRenderer, nullptr))
            invalidateVideoRenderer(*expiringVideoRenderer);
        return;
    }
```

this was your code, I reverted to it.